sktime/transformations/series/temporian.py

```diff
@@ -0,0 +1,123 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Implementation of TemporianTransformer. Contributed by the Temporian team."""
+
+__author__ = ["ianspektor"]  # TODO: add other Temporian devs who contributed to this
+
+from sktime.transformations.base import BaseTransformer
+
+
+# TODO: add usage example to docstring
+# see https://www.sktime.net/en/latest/developer_guide/dependencies.html#dependencies
+# for how to handle soft deps (temporian) there (need to not run doc tests on it)
+class TemporianTransformer(BaseTransformer):
+    """Applies a Temporian function to the input time series.
+
+    This transformer applies a [Temporian](https://temporian.readthedocs.io/en/stable/)
+    function to the input time series.
```

Comment:
@fkiraly, Temporian supports combining multiple sources of data, possibly with different sampling and indexing.
While the currently proposed code in sktime/transformations/series/temporian.py does only one-source-in one-source-out, multi-source support could be interesting in the future.

Does the Transformer abstraction support multi-source-in and/or multi-source-out?

Reply:
Depends. If you mean possibly hierarchically indexed collections of time series with different sampling rates etc, yes.

If you mean more complex relationally linked data tables, no.